framework/src/controller/application_state.js

```diff
@@ -0,0 +1,99 @@
+/* eslint-disable class-methods-use-this */
+/*
+ * Copyright © 2018 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+'use strict';
+
+const os = require('os');
+const crypto = require('crypto');
+const _ = require('lodash');
+
+/**
+ * Initial state of the entire application.
+ * - os
+ * - version
+ * - wsPort
+ * - httpPort
+ * - minVersion
+ * - protocolVersion
+ * - height
+ * - nethash
+ * - broadhash
+ * - nonce
+ *
+ * @class
+ * @requires crypto
+ * @requires os
+ * @param {Object} initialState - Initial state of the application
+ * @param {Object} logger
+ */
+class ApplicationState {
+	constructor(initialState, logger) {
+		this.logger = logger;
+		this.state = {
+			os: os.platform() + os.release(),
+			version: initialState.version,
+			wsPort: initialState.wsPort,
+			httpPort: initialState.httpPort,
```

Comment:
As we are already refactoring this whole concept, then we should reconsider this approach. `wsPort` and `httpPort` have no relation to application state and they must not be mentioned here.

Reply:
We will reconsider this in the P2P epic. Thanks for the reminder.